src/main/java/it/pagopa/pn/deliverypush/service/impl/SaveLegalFactsServiceImpl.java

```diff
@@ -99,11 +100,10 @@ public String saveNotificationReceivedLegalFact(NotificationInt notification) {
             log.debug("End saveNotificationReceivedLegalFact - iun={}", notification.getIun());
             logEvent.generateSuccess().log();
             return url;
-        }
-        catch ( Exception exc ) {
-            String msg = String.format(SAVE_LEGAL_FACT_EXCEPTION_MESSAGE, "REQUEST_ACCEPTED",  notification.getIun(), "N/A");
+        } catch (Exception exc) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)